pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -152,3 +167,130 @@ func (c configResource) computeAndReconcileConfigMap(xsite *configuration.XSite)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// newInfinispanOperatorConfigReconciler returns a new reconcile.Reconciler
+func newInfinispanOperatorConfigReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileInfinispanOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme(), Log: logf.Log.WithName("operator-config-controller"), Kube: kube.NewKubernetesFromController(mgr)}
+}
+
+var _ reconcile.Reconciler = &ReconcileInfinispanOperatorConfig{}
+
+// ReconcileInfinispan reconciles a Infinispan object
+type ReconcileInfinispanOperatorConfig struct {
+	client client.Client
+	scheme *runtime.Scheme
+	Log    logr.Logger
+	Kube   *kube.Kubernetes
+}
+
+// Reconcile reads the operator configuration in the `infinispan-operator-config` ConfigMap and
+// applies the configuration to the cluster.
+func (r *ReconcileInfinispanOperatorConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	err := r.reconcileOperatorConfigMap()
+	if err != nil {
+		r.Log.Error(err, "Error while computing and reconciling ConfigMap")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// reconcileOperatorConfigMap reconcile the operator ConfigMap
+// Operator configMap is unique for an operator install and placed in k8sutil.GetOperatorNamespace()/infinispan-operator-config
+func (r ReconcileInfinispanOperatorConfig) reconcileOperatorConfigMap() error {
+	config := &corev1.ConfigMap{}
+	operatorNs, err := getOperatorNamespace()
+	isRunningLocal := false
+	if err == k8sutil.ErrRunLocal {
+		isRunningLocal = true
+	}
+	if err != nil && !isRunningLocal {
+		return err
+	}
+	r.client.Get(ctx, types.NamespacedName{Namespace: operatorNs, Name: "infinispan-operator-config"}, config)
+	// Reconcile grafana
+	grafanaNs := config.Data["grafanaDashboardNameSpace"]
+	if grafanaNs != "" {
+		// detect the GrafanaDashboard resource type resourceExists on the cluster
+		resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+		if err != nil {
+			r.Log.Error(err, "Error checking Grafana support")
+		}
+		if !resourceExists {
+			r.Log.Info("Grafana CRD not present - not installing dashboard CR")
+			return nil
+		}
+		infinispanDashboard := grafana.Dashboard(grafanaNs)
+		operatorDeployment := &appsv1.Deployment{}
+		if !isRunningLocal {
+			result, err := infinispan.LookupResource("infinispan-operator", operatorNs, operatorDeployment, r.client, r.Log)
+			if err != nil {
+				r.Log.Error(err, "There was a problem looking up the infinispan operator to add the dashboard as a secondary resource")
+				return err
+			}
+			if result != nil {
+				r.Log.Info("Infinispan operator resource not initialized, not installing Grafana Dashboard")
+				return nil
+			}
+			// result and err should be analyzed
+			if err = controllerutil.SetControllerReference(operatorDeployment, infinispanDashboard, r.scheme); err != nil {
```

Comment:
I think it will be always fail when grafanaNs != operatorNs or will create a broken reference
Cross-namespace owner references are disallowed by design.

Reply:
mmm yeah, good catch. I'll set the controller only if ns match